cmd/snapd/main.go

```diff
@@ -102,6 +98,17 @@ func run() error {
 	if err := d.Init(); err != nil {
 		return err
 	}
+
+	// Run selftest now, if anything goes wrong with the selftest we go
+	// into "degraded" mode where we always report the given error to
+	// any snap client.
+	if err := selftest.Run(); err != nil {
+		degradedErr := fmt.Errorf("selftest failed: %s", err)
+		logger.Noticef("%s", degradedErr)
+		logger.Noticef("Entering degraded mode")
```

Comment:
Nitpick, it's easier to grep for if this is on one line: `logger.Noticef("entering degraded mode: %s", degradedErr)`